airflow/www/views.py

```diff
@@ -2564,6 +2562,8 @@ def action_clear(self, ids, session=None):
             # Collect dags upfront as dagbag.get_dag() will reset the session
             for id_str in ids:
                 task_id, dag_id, execution_date = id_str.split(',')
+                task_id = wwwutils.flask_admin_unescape(task_id)
```

Comment:
I wonder if there are other cases where this should be applied...

Reply:
Such as what about retrying tasks, or viewing task instance details. i.e. do we need to apply this much more globally to any/all flask admin views. (Or is there meant to be a different way of accessing these params when using flask admin?)

Reply:
We really should be doing this across all the flask_admin views that perform "actions" where the primary key of the SQL Alchemy model is a string. Right now in the UI there are only two views performing actions, one for "Task Instances" and another for "DAG Runs". The primary key for Task Instance includes dag_id (https://github.com/apache/incubator-airflow/blob/master/airflow/models.py#L771-L773) which has the periods enforced in within it whereas DAGRuns do not( https://github.com/apache/incubator-airflow/blob/master/airflow/models.py#L4563) which is why these are the only places I was able to identify that this was required. We could add that unescaping in the DAG Runs but it seemed unnecessary given that it's Integer id. I think there also may be places in the Tree or Graph View which may be affected by this issue, but I figured I would start with the known issue and then make another PR for that.